pkg/filesystem/local_directory_unix.go

```diff
@@ -460,3 +463,53 @@ func (d *localDirectory) Apply(arg interface{}) error {
 		return syscall.EXDEV
 	}
 }
+
+func (d *localDirectory) Mount(mountpoint path.Component, source string, fstype string, options int) error {
+	if options != 0 {
+		return status.Error(codes.InvalidArgument, "Options are not yet supported in `mountat`.")
+	}
+	mfd, err := mountat(d.fd, fstype, source, mountpoint.String())
+	if err != nil {
+		return util.StatusWrap(err, "Mountat")
+	}
+	defer unix.Close(mfd)
+
+	return nil
+}
+
+// Fchdir changes the current working directory to the directory file descriptor.
+// If there is an error, it will be of type *PathError.
+func Fchdir(dfd int) error {
+	if e := syscall.Fchdir(dfd); e != nil {
+		return &fs.PathError{Op: "fchdir", Path: fmt.Sprintf("%d", dfd), Err: e}
+	}
+	return nil
+}
+
+// Uses `unmount` on relative pathnames with `fchdir`.
+func fchdir_unmountat(lock *sync.Mutex, dfd int, mountname string) error {
+	lock.Lock()
+	defer lock.Unlock()
+
+	store, err := os.Getwd()
+	if err != nil {
+		return util.StatusWrap(err, "Could not store the working directory %#v.")
+	}
+	defer os.Chdir(store)
```

Comment:
I'd say, don't even bother restoring the working directory. If other pieces of code depend on doing relative path lookups, then they will be broken anyway because they race with this code.

That said, I can imagine it is desirable to leave the FUSE/NFSv4 file system, so that it doesn't remain in use permanently. What are your thoughts on doing an `os.Chdir("/")` instead?

Reply:
I feel a bit uneasy about leaving it in whereverland, but going to the root is fine.
When I debugged this code path it seems to live in /tmp and goes back there after unmounting.

Reply:
Using the old `unmount` now that does not attempt to return.